### PR TITLE
CLI: Fix failure on dir with trailing slash

### LINKED ISSUE #9300: Cli different between `prettier .` and `prettier ./`
I found this long time ago, but fogot to open an issue. 

I forgot what's the difference between them. I'm not near a desktop, will check later.

### LINKED ISSUE #10964: CLI fails when passed a dir with trailing slash: `prettier dir/`
**Environments:**

- Prettier Version: 2.3.0
- Running Prettier via: CLI
- Runtime: Node.js v14
- Operating System: Linux (WSL 2 w/ Debian buster)
- Prettier plugins (if any): No plugins

**Steps to reproduce:**

Run `npx prettier code/` with a trailing slash, **not** `npx prettier code`.

The problem is that many tools, including bash on Debian, add a trailing slash to directory names entered on the prompt **when** they get completed pressing TAB, instead of becoming fully typed out manually.

**Expected behavior:**

It should have printed the formatted files from the specified `code` directory recursively.

**Actual behavior:**

This Error gets logged:
`[error] No supported files were found in the directory: "code/".`

### PR DESCRIPTION
**Pull request by fisker**
## Description

Fixes #9300
Fixes #10964 

## Checklist

- [ ] I’ve added tests to confirm my change works.
- [ ] (If changing the API or CLI) I’ve documented the changes I’ve made (in the `docs/` directory).
- [ ] (If the change is user-facing) I’ve added my changes to `changelog_unreleased/*/XXXX.md` file following `changelog_unreleased/TEMPLATE.md`.
- [ ] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/main/CONTRIBUTING.md).

**✨[Try the playground for this PR](https://prettier.io/playground-redirect)✨**

### CODE REVIEW

File: src/cli/expand-patterns.js [hunk 1]
```diff
@@ -76,10 +76,11 @@ async function* expandPatternsInternal(context) {
           input: pattern,
         });
       } else if (stat.isDirectory()) {
+        const relativePath = path.relative(cwd, absolutePath) || ".";
```
Comment (thorn0) [inline src/cli/expand-patterns.js:79]: If the goal of this line is only to remove trailing slashes, why not do `.replace(/[/\\]+$/, '')` here to express the intent more clearly?

Comment (fisker) [reply src/cli/expand-patterns.js:79]: I think the middle one `foo//bar` was effected too.

**Review by thorn0 — APPROVED** [review]

File: src/cli/expand-patterns.js (hunk 1 above)
Comment (fisker) [reply src/cli/expand-patterns.js:79]: I'll confirm when I back to laptop, if it's true, I'll add comment.

Comment (fisker) [reply src/cli/expand-patterns.js]: Tested, the tailing slash is the only problem, applied your suggestion.

Comment (fisker) [reply src/cli/expand-patterns.js:84]: Your solution can't work `\` can be the dirname, see this log https://github.com/prettier/prettier/runs/2746762536#step:6:1251

This commit https://github.com/prettier/prettier/pull/11000/commits/da461964378c29df112f67d7f44a52f1147e8955 works, but feel better/safer to use `path.relative`,  added comments https://github.com/prettier/prettier/pull/11000/commits/b00987886d90edaf9d34b3e45b56fd56ab659ebd

**Review by sosukesuzuki — APPROVED** [review]

**Pull request merged**